baby-gru/src/components/validation-tools/MoorhenCarbohydrateValidation.tsx

```diff
@@ -0,0 +1,182 @@
+            ')';
+
+        props.clickCallback(newCenterString);
+        const selectedMolecule = molecules.find(
+            (molecule) => molecule.molNo === props.selectedModel
+        );
+        const _center = await selectedMolecule.centreOn(newCenterString);
+    }
+
+    const sugarRef = useCallback(
+        (node: HTMLElement | null) => {
+            if (node !== null) {
+                node.querySelector('svg').style.display = 'block';
+                node.querySelector('svg').style.margin = 'auto';
+
+                const useList = node.querySelectorAll('use');
+
+                for (let i = 0; i < useList.length; i++) {
+                    useList[i].addEventListener('click', handleClick);
+                    useList[i].addEventListener('mousedown', (e) => {
+                        e.stopPropagation();
+                    });
+                    useList[i].addEventListener('touchstart', (e) => {
+                        e.stopPropagation();
+                    });
+                }
+            }
+        }, []
+    );
+
+    return <>
+        {props.data.map((item, index) => {
+            return (
+                <div style={{padding: "0 0 1rem 1rem", borderBottom: "solid"}}>
+                    <div style={{display: "flex", justifyContent: "space-between", padding: "1rem 1rem 0rem 1rem"}}>
+                    <p>Glycan ID: {item.id}</p>
+                        <p>GlyToucan ID: {item.glytoucan_id}</p>
+                            <p>GlyConnect ID: {item.glyconnect_id}</p>
+                    </div>
+                <div
+                    style={{display: "flex", padding: "1rem"}}
+                    id="svgContainer"
+                    dangerouslySetInnerHTML={{
+                        __html: item.svg,
+                    }}
+                    ref={sugarRef}
+                />
+                </div>
+            )
+        })}
+
+        <p style={{margin: "auto"}}><i>Powered by Privateer</i></p>
+    </>
+
+}
+
+export const MoorhenCarbohydrateValidation = (props: Props) => {
+
+    const isDark = useSelector((state: moorhen.State) => state.sceneSettings.isDark)
+    const mapSelectRef = useRef<undefined | HTMLSelectElement>();
+    const moleculeSelectRef = useRef<undefined | HTMLSelectElement>();
+
+    const molecules = useSelector((state: moorhen.State) => state.molecules)
+    const maps = useSelector((state: moorhen.State) => state.maps)
+    const [selectedModel, setSelectedModel] = useState<null | number>(null)
+    const [selectedMap, setSelectedMap] = useState<null | number>(null)
+
+    const [privateerData, setPrivateerData] = useState<null | PrivateerResultsEntry[]>(null)
+
+
+    const handleModelChange = (evt: React.ChangeEvent<HTMLSelectElement>) => {
+        setSelectedModel(parseInt(evt.target.value))
+    }
+
+    const handleMapChange = (evt: React.ChangeEvent<HTMLSelectElement>) => {
+        setSelectedMap(parseInt(evt.target.value))
+    }
```

Comment:
Again this widget does not use map data (I think?) so please remove `handleMapChange`